### PR TITLE
Integrate otj-server with the logging k8s changes

### PR DESCRIPTION
**Pull request by mikebell90**
Requires
* 1.3.11 otj-spring (fix regression)
* 3.0.11 otj-logging actual integration

Notes
=====

1. A test is failing. I don't think it has to do with this PR, but obviously needs to be fixed. (See Jaxrs)

2. Needs testing in both K8s and Singularity

3. @adamzr  @dkaukov  dunno if you have a better place to inject - the problem was the ServerLoggingConfiguration is now a Spring factory, and injection doesn't seem to work, probably because the Context isn't completely ready there. This means there's some time that can pass before the K8s stuff gets injected, which is certainly unfortunate.

### CODE REVIEW

**Comment by mikebell90** [comment]
Source of test failure has been identified. Indeed it has nothing to do with this PR, but with the newer POM using ReferringInformation from otj-spring. This was my attempt to consolidate some information.

Basically, Chris's test shows JAXRS client filter doesn't register if no headers are registered. Now there is one.

However the issue is primarily caused by very minimal unit tests that didn't inject AppInfo or ServiceInfo. I think the breakage is acceptable and people will fix their tests - it's QUITE complicated to redo all. those PRs and have 3 chains of optionals for this case.

Opinions? Mine is just delete the test, and let people fix their tests. I don't expect wide spread damage.

@enorrbinot This was originally OTPL-1961, inspired by you. Your opinion?

**Comment by enorrbinot** [comment]
I don't remember the exact tests that failed, but I suspect it was some of the client unit tests in Promoted Inventory service. I'm sure we can fix those if needed.

File: otj-server-jaxrs/src/test/java/com/opentable/jaxrs/referrer/JaxrsReferrerMinimalTest.java [hunk 1]
```diff
@@ -38,6 +39,7 @@
     private ClientReferrerFilter filter;
 
     @Test
+    @Ignore //TODO: broken for some reason
     public void test() {
         Assertions.assertThat(filter.isActive()).isFalse();
     }
```
Comment (rushalias) [inline otj-server-jaxrs/src/test/java/com/opentable/jaxrs/referrer/JaxrsReferrerMinimalTest.java]: based on your findings & suggestion this test will be deleted?

Comment (mikebell90) [reply otj-server-jaxrs/src/test/java/com/opentable/jaxrs/referrer/JaxrsReferrerMinimalTest.java]: yes

**Pull request merged**